config/config_params.go

```diff
@@ -191,6 +193,7 @@ type Config struct {
 	HealthPort                      int    `config:"int(0,65535);9099"`
 	HealthHost                      string `config:"string;localhost"`
 	PrometheusMetricsEnabled        bool   `config:"bool;false"`
+	PrometheusMetricsHost           string `config:"prometheus-host;"`
```

Comment:
Why should the validation for PrometheusMetricsHost be different from that for HealthHost?
- If there is a good reason for the different handling, please add a comment in the code about that.
- If the `prometheus-host` handling is no better than what we already do for HealthHost, please revert the prometheus-host changes and instead do the same as we do for HealthHost.
- If the `prometheus-host` handling is clearly better than what we do for other host config parms, please rename to make it non-prometheus-specific, and apply to all similar parameters.

Reply:
I see that `HealthHost` currently has no validation eventually ends up in the same net/http/server.go, but before that there is a difference in how the data ends up there:
`fmt.Sprintf("%s:%v", host, port)` in libcalico.go
vs
`fmt.Sprintf("[%s]:%v", host, port)` // in typha and felix project daemon.go

My validation makes sure that when you enter the value for prometheushost, you are not including '[]', which makes sure that ipv4, ipv6 and 'localhost' would all work. If I apply it to HealthHost, ipv6 would not work unless I modify libcalico.go accordingly. If I change the validation, I would require more logic to make sure the input value works.
Want me to create a separate PR for HealthHost in libcalico.go?

Reply:
So you're saying that the validation for Prometheus is clearly better, right?  In that case I think we should apply it everywhere.

Reply:
Ok, please see also: 
projectcalico/typha#285
projectcalico/libcalico-go#1109